src/types.rs

```diff
@@ -376,7 +387,42 @@ impl Storable for Provider {
 }
 
 impl BoundedStorable for Provider {
-    const MAX_SIZE: u32 = 256; // A reasonable limit.
+    const MAX_SIZE: u32 = PROVIDER_MAX_SIZE;
+    const IS_FIXED_SIZE: bool = false;
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub struct StorableRpcService(Vec<u8>);
+
+impl TryFrom<StorableRpcService> for RpcService {
+    type Error = serde_json::Error;
+    fn try_from(value: StorableRpcService) -> Result<Self, Self::Error> {
+        serde_json::from_slice(&value.0)
+    }
+}
+
+impl StorableRpcService {
+    pub fn new(service: &RpcService) -> Self {
+        // Store as JSON string for enum order invariance
```

Comment:
Hopefully this is sufficiently explained by the comment, but the reason for using JSON serialization here is so that there's essentially no way for the RPC services to accidentally point to each other if we make any changes to the `RpcService`, `EthMainnetService`, or `EthSepoliaService` enums down the road.

Reply:
This is a good explanation but the comment only mentions "enum order invariance".

Reply:
Updated. I preferred the original, more concise comment, but I suppose it's other people reading it. 😄